lib/rspec/rails/matchers/active_job.rb

```diff
@@ -89,7 +96,7 @@ def supports_block_expectations?
         private
 
           def check(jobs)
-            @matching_jobs_count = jobs.count do |job|
+            @matching_jobs, @unmatching_jobs = jobs.partition do |job|
```

Comment:
Yes, it will return all jobs that were queued during given action.
I think most of the time number of jobs should be low, so we don't need to scope only to the same class jobs.
This way it's clear for end user what's going on, and it may happen that mistake is also in job class name (like copy pasting) ;-)